homeassistant/components/sensorpush/data.py

```diff
@@ -0,0 +1,87 @@
+"""The SensorPush Bluetooth integration."""
+from __future__ import annotations
+
+from sensorpush_ble import (
+    ATTR_MANUFACTURER as SENSOR_MANUFACTURER,
+    ATTR_MODEL as SENSOR_MODEL,
+    ATTR_NAME as SENSOR_NAME,
+    SIGNAL_STRENGTH_KEY,
+    DeviceClass,
+    DeviceKey,
+    SensorDeviceInfo,
+    SensorUpdate,
+)
+
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntityDescription
+from homeassistant.const import ATTR_MANUFACTURER, ATTR_MODEL, ATTR_NAME
+from homeassistant.helpers.entity import DeviceInfo
+
+SENSOR_DEVICE_CLASS_TO_HASS = {
```

Comment:
I wonder if you can generate this mapping with a duct comprehension and if that would be shorter.

Reply:
Kinda meh but it would work with

```python
SENSOR_DEVICE_CLASS_TO_HASS = {
    getattr(DeviceClass, key): getattr(SensorDeviceClass, key)
    for key in ["BATTERY", "HUMIDITY", "TEMPERATURE", "PRESSURE", "SIGNAL_STRENGTH"]
}
```

Reply:
Maybe slightly better
```python
SENSOR_DEVICE_CLASS_TO_HASS = {
    key: getattr(SensorDeviceClass,key.value)
    for key in DeviceClass if hasattr(SensorDeviceClass,key.value)
}
```